system/Database/SQLSRV/Connection.php

```diff
@@ -511,7 +512,7 @@ public function execute(string $sql)
 	 */
 	public function isWriteType($sql)
 	{
-		return (bool) preg_match('/^\s*"?(SET|INSERT|UPDATE|DELETE|REPLACE|CREATE|DROP|TRUNCATE|LOAD|COPY|ALTER|RENAME|GRANT|REVOKE|LOCK|UNLOCK|REINDEX|MERGE)\s/i', $sql);
+		return Query::sqlIsWriteType($sql);
```

Comment:
It seems it is a bad idea that moving a driver specific logic to common  `Query::sqlIsWriteType()`.
The regex is specific for the driver, not all the same.